recommendation/widedeep/widedeep.py

```diff
@@ -0,0 +1,123 @@
+            parsed_feature_vals = parsed_feature_vals_num + parsed_feature_vals_str
+
+            # labels
+            parsed_feature_vals_label = [tf.reshape(parsed_features[i], shape=[-1]) for i in ["label"]]
+            parsed_feature_vals = parsed_feature_vals + parsed_feature_vals_label
+
+            return parsed_feature_vals
+
+        # Extract lines from input files using the Dataset API.
+        dataset = tf.data.TFRecordDataset([self.tfrecords_path])
+        if shuffle:
+            dataset = dataset.shuffle(buffer_size=20000)
+        dataset = dataset.batch(self.batch_size)
+        # If the value tf.data.AUTOTUNE is used, then the number of parallel calls is set dynamically based on CPU.
+        dataset = dataset.map(_parse_function, num_parallel_calls=tf.data.AUTOTUNE)
+        dataset = dataset.prefetch(tf.data.AUTOTUNE)
+        return dataset
+
+    def get_features_list(self, config, graph, no_of_batches):
+        features_list = []
+        with tf.compat.v1.Session(config=config, graph=graph) as sess:
+            res_dataset = self.input_fn(False)
+            iterator = tf.compat.v1.data.make_one_shot_iterator(res_dataset)
+            next_element = iterator.get_next()
+            for _ in range(no_of_batches):
+                batch = sess.run(next_element)
+                features_list.append(batch)
+
+        return features_list
+
+    def unpickle(self, dataset_path):
+        file = bz2.BZ2File(dataset_path, 'rb')
+        features_list = pickle.load(file)
+        file.close()
+        return features_list
+
+    def get_input_array(self):
+
+        try:
+            # self.features_list[self.current_feature] is a tuple object containing input sample and ground truth
+            # :2 indexing is used to extract just the input sample and index 2 contains ground truths.
+            input_array = self.features_list[self.current_feature][:2]
+        except IndexError:
+            raise utils.OutOfInstances("no more features to process")
+        return input_array
+
+    def reset(self):
+        self.current_feature = 0
+        self.correct = 0
+        return True
+
+    def submit_predictions(self, output_array):
+        top_1 = np.argmax(output_array['import/import/head/predictions/probabilities:0'], 1)
+        # self.features_list[self.current_feature] is a tuple object containing input sample and ground truth
+        # :2 indexing is used to extract just the input sample and index 2 contains ground truths.
+        self.correct += np.sum(self.features_list[self.current_feature][2] == top_1)
+        self.current_feature += 1
+
+    def summarize_accuracy(self):
+        run_instances = self.current_feature * self.batch_size
+        accuracy = float(self.correct) / float(run_instances)
+
+        print("accuracy = {:.3f}".format((accuracy * 100), 4))
+        print("correct predictions = {:.3f}".format(self.correct))
+        print("total predictions = {:.3f}".format(run_instances))
```

Comment:
you should return dict with accuracy metric for model_zoo_dev facilities

Reply:
- formatting could be better, new line before "accuracy...", some indent
- why display predictions count with >0 decimal places?

that's how it looks now:
```
WARNING:tensorflow:From /ampere/ampere_model_library/recommendation/widedeep/widedeep.py:28: tf_record_iterator (from tensorflow.python.lib.io.tf_record) is deprecated and will be removed in a future version.
Instructions for updating:
Use eager execution and:
`tf.data.TFRecordDataset(path)`
2023-05-23 22:04:22.645762: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:354] MLIR V1 optimization pass is not enabled
accuracy = 78.379
correct predictions = 16422.000
total predictions = 20952.000

                               mean  /      median  / 90th-percentile
 Latency           [ms]:       0.40  /        0.40  /        0.43
 Throughput [samples/s]:    2473.08  /     2513.06  /     2340.57

```